x-pack/plugins/session_view/public/components/process_tree_alerts/index.tsx

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import {
+  EuiButton,
+  EuiText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHorizontalRule,
+  EuiSpacer,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useStyles } from './styles';
+import { ProcessEvent } from '../../../common/types/process_tree';
+import { useKibana } from '../../../../../../src/plugins/kibana_react/public';
+import { CoreStart } from '../../../../../../src/core/public';
+
+interface ProcessTreeAlertsDeps {
+  alerts: ProcessEvent[];
+}
+
+export function ProcessTreeAlerts({ alerts }: ProcessTreeAlertsDeps) {
+  const styles = useStyles();
+  const { http } = useKibana<CoreStart>().services;
+
+  if (alerts.length === 0) {
+    return null;
+  }
+
+  const getRuleUrl = (alert: ProcessEvent) => {
+    return http.basePath.prepend(`/app/security/rules/id/${alert.kibana?.alert.rule.uuid}`);
+  };
+
+  const renderAlertDetails = (alert: ProcessEvent, index: number) => {
```

Comment:
I think almost all of these const renderWhatever functions should be rewritten as components, as they are just functions that return jsx, but unless the function starts with an uppercase letter to denote it's a component or useWhatever to denote it's a custom hook, it'll get rerun on every render, could lead to bugs if it's a child of a component that makes heavy use of useState/useEffect etc.